### ISSUE TITLE
Cannot complete build

### ISSUE THREAD

**Issue by stavros-k** [dc7290/next-export-optimize-images#696]
**Describe the bug**

Hello, I was trying to use package to optimize my images. But after following the getting started guide and trying to build, I get the following error.

```shell
❯ npm run build                    

> test@0.1.0 build
> next build

info - [next-export-optimize-images]: Configuration was not loaded. (This is optional.)
 ✓ Creating an optimized production build    
 ✓ Compiled successfully
 ✓ Linting and checking validity of types    
 ✓ Collecting page data    
   Generating static pages (0/5)  [=   ]TypeError: Cannot read properties of null (reading 'useContext')
    at exports.useContext (/tmp/test/node_modules/react/cjs/react.production.min.js:24:118)
    at /tmp/test/node_modules/next/dist/client/image-component.js:241:47
    at nh (/tmp/test/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:27:48668)
    at ny (/tmp/test/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:27:52524)
    at nf (/tmp/test/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:27:35987)
    at nh (/tmp/test/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:27:48675)
    at nh (/tmp/test/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:27:49347)
    at ny (/tmp/test/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:27:52524)
    at nv (/tmp/test/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:27:55162)
    at ng (/tmp/test/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:27:54569)
```

**To Reproduce**

1. Create a new project.
```shell
❯ npx create-next-app@latest
Need to install the following packages:
create-next-app@13.5.6
Ok to proceed? (y) y
✔ What is your project named? … test
✔ Would you like to use TypeScript? … No / **Yes**
✔ Would you like to use ESLint? … No / **Yes**
✔ Would you like to use Tailwind CSS? … No / **Yes**
✔ Would you like to use `src/` directory? … **No** / Yes
✔ Would you like to use App Router? (recommended) … No / **Yes**
✔ Would you like to customize the default import alias (@/*)? … **No** / Yes
Creating a new Next.js app in /tmp/test.

Using npm.
```
2. Configure output for export
```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  output: "export",
};

module.exports = nextConfig;
```
3. Run `npm run build` and observe succesful build
4. Install the `next-export-optimize-images` package
5. Modify `next.config.js`
```js
const withExportImages = require("next-export-optimize-images");

/** @type {import('next').NextConfig} */
const nextConfig = withExportImages({
  output: "export",
});

module.exports = nextConfig;
```
6. Try to build, you will see the error

**Expected behavior**

A successful build.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Linux/Ubuntu
 - Browser [e.g. chrome, safari] N/A
 - Version [e.g. 22] N/A

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone6] N/A
 - OS: [e.g. iOS8.1] N/A
 - Browser [e.g. stock browser, safari] N/A
 - Version [e.g. 22] N/A

**Additional context**

I'm pretty new to next.js, been trying it for few days only, so it might be user error.
I didnt't see any other means to start a discussion, which is why I opened this bug report.

Thanks

**Comment by darthrommy** [comment]
I'm also getting the same error, but I found a temporary workaround: **fall back to `pages` router**. (`app` router is enabled by default)

I tried switching to `pages` router because I doubted that the new `app` router had something to do with it, and then this plugin worked fine without any errors.

Since the README says "AppRouter Support", I think this is not your error but a bug.

**Comment by tiste** [comment]
Same error: `TypeError: Cannot read properties of null (reading 'useContext`

**Comment by tiste** [comment]
Just reproduced it with the actual `examples/common`.

Rename `pages` folder to `app`, and add the required `RootLayout`:

```js
export default function RootLayout({ children }) {
  return (
    <html>
      <body>
        {children}
      </body>
    </html>
  );
}
```

Then the error appear.

**Comment by tiste** [comment]
@stavros-k just succeed by downgrading next from latest `13.5.6` to `13.4.19`.
Seems to have a bug on it: https://github.com/vercel/next.js/issues/57277

**Comment by stavros-k** [comment]
Same on Next 14. (I had to enable experimental webpack builder)
```
    experimental: {
        webpackBuildWorker: true,
    },
```

**Comment by tiste** [comment]
> Same on Next 14. (I had to enable experimental webpack builder)
> 
> ```
>     experimental: {
>         webpackBuildWorker: true,
>     },
> ```

@stavros-k I still have issues with next 14 even with this option :(

**Comment by stavros-k** [comment]
> > Same on Next 14. (I had to enable experimental webpack builder)
> > ```
> >     experimental: {
> >         webpackBuildWorker: true,
> >     },
> > ```
> 
> @stavros-k I still have issues with next 14 even with this option :(

Oh, yea should have been clearer.
It's still an issue, the webpack option was just an extra thing that next was asking to enable in order to "continue". But it still fails with the same `useContext` error!

I'll stick with 14 for now, until the fix come from either side, we already had unoptimized images for ever, so it won't hurt for few more days/weeks.

**Comment by J3tto** [comment]
Any update on when/if this will be resolved?

**Comment by joebentaylor1995** [comment]
No fixes for this still?

**Comment by dc7290** [comment]
https://github.com/dc7290/next-export-optimize-images/issues/796

**Comment by dc7290** [comment]
Ref: https://github.com/dc7290/next-export-optimize-images/issues/820

**Comment by dc7290** [comment]
Fixed release in v4.0.0!
